sklearn/gaussian_process/_gpr.py

```diff
@@ -519,3 +519,20 @@ def _constrained_optimization(self, obj_func, initial_theta, bounds):
 
     def _more_tags(self):
         return {'requires_fit': False}
+
+
+def _handle_zeros_in_std(std, copy=True):
```

Comment:
Why not rather import existing?
https://github.com/scikit-learn/scikit-learn/blob/27f1c737f2cfc6e589c292776df6f78c08a4da91/sklearn/preprocessing/tests/test_data.py#L49

Reply:
@kernc
I thought it would be "unconventional" to import function with underscore prefix. I'll fix this.

Reply:
Ideally, the function would be rephrased as `return np.where(std, std, 1, copy=copy)` and phased out.

Reply:
Done. 
It seems that `np.where()` doesn't have an argument `copy`, please, let me know if I'm missing something

Reply:
It doesn't. I said ideally. :sweat_smile:

You are aware I'm not a maintainer here? With the amount of change now, this might have fewer chances to be accepted. Change is expensive. Not to mention how `np.where()` always produces a copy, which might be less efficient. Best get some maintainer feedback first.

Reply:
> You are aware I'm not a maintainer here?

Sorry, I've just noticed that :woman_facepalming:

You are right, for some cases additional resource consumption caused by `np.where()` may appear to be crucial, so it seems logical not to touch all parts that work correct and return to the usage of  `_handle_zeros_in_scale` with `copy=False` in GaussianProcessRegressor